clang/lib/CodeGen/CGAtomic.cpp

```diff
@@ -766,8 +766,17 @@ static void EmitAtomicOp(CodeGenFunction &CGF, AtomicExpr *Expr, Address Dest,
   // LLVM atomic instructions always have synch scope. If clang atomic
   // expression has no scope operand, use default LLVM synch scope.
   if (!ScopeModel) {
+    llvm::SyncScope::ID SS = CGF.getLLVMContext().getOrInsertSyncScopeID("");
+    if (CGF.getLangOpts().OpenCL)
+      // OpenCL approach is: "The functions that do not have memory_scope
```

Comment:
Which flavor of atomic operations does this function correspond to?

Reply:
This is the primary entry point for Atomic emission, so things like the Clang builtins (which do not carry scopes) would end up here.